### PR TITLE
[ENH] Add MLPRegressor 

### LINKED ISSUE #1217: `MLPRegressor`

### PR DESCRIPTION
**Pull request by aadya940**
#### Reference Issues/PRs

Related to #1198 
Fix #1217 

#### What does this implement/fix? Explain your changes.

1. Create _mlp.py Module in aeon.regression
2. Call MLPNetwork
3. Replace Ouput-Layer Activation by "linear"
4. Update Loss to "mse"
5. Update docstrings

### CODE REVIEW

File: aeon/regression/deep_learning/_mlp.py [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+            metrics=metrics,
+        )
+        return model
+
+    def _fit(self, X, y):
+        """Fit the Regressor on the training set (X, y).
+
+        Parameters
+        ----------
+        X : np.ndarray of shape = (n_instances (n), n_channels (d), series_length (m))
```
Comment (hadifawaz1999) [inline aeon/regression/deep_learning/_mlp.py:176]: could you replace the information to (n_instances, n_channels, n_timepoints)

File: aeon/regression/deep_learning/_mlp.py [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+            The training input samples.
+        y : np.ndarray of shape n
+            The training output data.
```
Comment (hadifawaz1999) [inline aeon/regression/deep_learning/_mlp.py:179]: here it should be "The training data target values" for the y parameter

File: aeon/regression/deep_learning/_mlp.py [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+from copy import deepcopy
+
+from sklearn.utils import check_random_state
+
+from aeon.networks import MLPNetwork
+from aeon.regression.deep_learning.base import BaseDeepRegressor
+
+
+class MLPRegressor(BaseDeepRegressor):
+    """Multi Layer Perceptron Network (MLP).
```
Comment (hadifawaz1999) [inline aeon/regression/deep_learning/_mlp.py:18]: can you re-arrange the docstring so that the order of the documentation of each parameter is in the same order as they are listed inside __init__ ?

**Review by hadifawaz1999 — CHANGES_REQUESTED** [review]
thanks for taking care of this, just a few comments

**Comment by aadya940** [comment]
@hadifawaz1999 The docstring for `callback` is not present in both `MLPClassifier` as well as `MLPRegressor`. Should I add it to both?

File: aeon/regression/deep_learning/_mlp.py
[the remark this answers is not in the document]
Comment (aadya940) [reply aeon/regression/deep_learning/_mlp.py:1]: @AnonymousCodes911 No, the reason being `convert_y_to_keras` converts `y` to a One-Hot Array that is a Mutually Exclusive Array of `y` values typically used for categorical data in classification. We don't need it in Regression!

**Comment by hadifawaz1999** [comment]
> @hadifawaz1999 The docstring for `callback` is not present in both `MLPClassifier` as well as `MLPRegressor`. Should I add it to both?

Yes it would be great if you add it thanks ! could you also take care of the comments i left above ?

**Review by hadifawaz1999 — CHANGES_REQUESTED** [review]
can you merge with main to apply the network module changes for the random_state parameter ?

**Comment by aadya940** [comment]
@hadifawaz1999 I've added docstrings for `callbacks` in `regression._mlp` as well as `classification._mlp` as well as merged the changes for `random_state` and also resolved the review comments. Thanks :-)

**Review by hadifawaz1999 — APPROVED** [review]
LGTM! thanks
